### PR TITLE
fix(tsm1): tsmFullCompactionQueue has not been updated ever

### PR DESCRIPTION
**Pull request by StoneYunZhao**
tsmFullCompactionQueue has not been updated ever. 

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: tsdb/engine/tsm1/cache.go [hunk 1]
```diff
@@ -336,7 +336,7 @@ func (c *Cache) WriteMulti(values map[string][]Value) error {
 	store := c.store
 	c.mu.RUnlock()
 
-	// We'll optimistially set size here, and then decrement it for write errors.
+	// We'll optimistically set size here, and then decrement it for write errors.
 	c.increaseSize(addedSize)
```
Comment (StoneYunZhao) [inline tsdb/engine/tsm1/cache.go:340]: Just typo.

File: tsdb/engine/tsm1/compact.go [hunk 2]
```diff
@@ -834,7 +834,7 @@ func (c *Compactor) WriteSnapshot(cache *Cache) ([]string, error) {
 	// Enable throttling if we have lower cardinality or snapshots are going fast.
 	throttle := card < 3e6 && c.snapshotLatencies.avg() < 15*time.Second
 
-	// Write snapshost concurrently if cardinality is relatively high.
+	// Write snapshot concurrently if cardinality is relatively high.
 	concurrency := card / 2e6
```
Comment (StoneYunZhao) [inline tsdb/engine/tsm1/compact.go:838]: Just typo.

File: tsdb/engine/tsm1/engine.go [hunk 3]
```diff
@@ -2047,15 +2047,14 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 	}
 }
 
-// compactHiPriorityLevel kicks off compactions using the high priority policy. It returns
+// compactLevel kicks off compactions using the level strategy. It returns
 // true if the compaction was started
-func (e *Engine) compactHiPriorityLevel(grp CompactionGroup, level int, fast bool, wg *sync.WaitGroup) bool {
+func (e *Engine) compactLevel(grp CompactionGroup, level int, fast bool, wg *sync.WaitGroup) bool {
 	s := e.levelCompactionStrategy(grp, fast, level)
```
Comment (StoneYunZhao) [inline tsdb/engine/tsm1/engine.go:2053]: Method compactLoPriorityLevel and method compactHiPriorityLevel code are exactly the same.

File: tsdb/engine/tsm1/engine.go [hunk 4]
```diff
@@ -1997,7 +1997,7 @@ func (e *Engine) compact(wg *sync.WaitGroup) {
 			level2Groups := e.CompactionPlan.PlanLevel(2)
 			level3Groups := e.CompactionPlan.PlanLevel(3)
 			level4Groups := e.CompactionPlan.Plan(e.LastModified())
-			atomic.StoreInt64(&e.stats.TSMOptimizeCompactionsQueue, int64(len(level4Groups)))
+			atomic.StoreInt64(&e.stats.TSMFullCompactionsQueue, int64(len(level4Groups)))
```
Comment (StoneYunZhao) [inline tsdb/engine/tsm1/engine.go:2001]: Here is related to PR.

**Review by danxmoran — COMMENTED** [review]
Apologies for the delay in reviewing this.

Can you split this into 2 separate PRs? One for `engine.go`, and one for `compact.go` + `cache.go`. I feel most confident about the `engine.go` changes; the `compact.go` change looks mostly ok to me, but I need some more time to dig around in the code to make sure I'm not missing something.

**Comment by StoneYunZhao** [comment]
> Apologies for the delay in reviewing this.
> 
> Can you split this into 2 separate PRs? One for `engine.go`, and one for `compact.go` + `cache.go`. I feel most confident about the `engine.go` changes; the `compact.go` change looks mostly ok to me, but I need some more time to dig around in the code to make sure I'm not missing something.

I updated this PR to be only related to `engine.go` except for typo.

**Comment by StoneYunZhao** [comment]
> Apologies for the delay in reviewing this.
> 
> Can you split this into 2 separate PRs? One for `engine.go`, and one for `compact.go` + `cache.go`. I feel most confident about the `engine.go` changes; the `compact.go` change looks mostly ok to me, but I need some more time to dig around in the code to make sure I'm not missing something.

As for `getXXXEncoder(tsdb.DefaultMaxPointsPerBlock)` in `compact.go`, I think it’s OK if don’t change it, because:
1. `c.size` is hard-coded in the code and is equal to `tsdb.DefaultMaxPointsPerBlock`.
2. `getXXXEncoder(size)` only affects memory allocation, not program logic errors.
3. `tsdb.DefaultMaxPointsPerBlock` is hard-coded in many places.

**Review by danxmoran — APPROVED** [review]
Sorry it took so long, LGTM!

**Pull request merged**